### PR TITLE
xandikos: 0.2.8 -> 0.2.10

### PR DESCRIPTION
**Pull request by schnusch**
## Description of changes

Diff: https://github.com/jelmer/xandikos/compare/v0.2.8...v0.2.10

Changelog: https://github.com/jelmer/xandikos/blob/v0.2.10/NEWS

Depends on <https://github.com/NixOS/nixpkgs/pull/253611>.

`prometheus-client` was previously replaced by `aiohttp-openmetrics`, see <https://github.com/jelmer/xandikos/commit/d5da74ad098b6fc42462a1c3974c229dcb7a35de>.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [x] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/servers/xandikos/default.nix [hunk 1]
```diff
@@ -1,38 +1,71 @@
...
+  format = "pyproject";
+
+  disabled = pythonOlder "3.9";
 
   src = fetchFromGitHub {
     owner = "jelmer";
     repo = "xandikos";
     rev = "v${version}";
-    sha256 = "sha256-KDDk0QSOjwivJFz3vLk+g4vZMlSuX2FiOgHJfDJkpwg=";
+    sha256 = "sha256-SqU/K3b8OML3PvFmP7L5R3Ub9vbW66xRpf79mgFZPfc=";
```
Comment (0x4A6F) [inline pkgs/servers/xandikos/default.nix:32]: Could migrate to hash here.

```diff
-    sha256 = "sha256-KDDk0QSOjwivJFz3vLk+g4vZMlSuX2FiOgHJfDJkpwg=";
+    hash = "sha256-SqU/K3b8OML3PvFmP7L5R3Ub9vbW66xRpf79mgFZPfc=";
```

File: pkgs/servers/xandikos/default.nix [hunk 2]
```diff
@@ -1,38 +1,71 @@
...
   passthru.tests.xandikos = nixosTests.xandikos;
 
-  nativeCheckInputs = with python3Packages; [ pytestCheckHook ];
+  nativeCheckInputs = [ pytestCheckHook ];
   disabledTests = [
     # these tests are failing due to the following error:
-    # TypeError: expected str, bytes or os.PathLike object, not int
-    "test_iter_with_etag"
-    "test_iter_with_etag_missing_uid"
+    # TypeError: index_entry_from_stat() missing 1 required positional argument
```
Comment (natsukium) [inline pkgs/servers/xandikos/default.nix:56]: I think something is wrong.
Do these tests also fail upstream?

Comment (schnusch) [reply pkgs/servers/xandikos/default.nix:56]: I disabled those before I saw dulwich needed updating. With dulwich 0.21.6 its no longer necessary.

**Review by kip93 — APPROVED** [review]

File: pkgs/top-level/all-packages.nix [hunk 3]
```diff
@@ -25742,7 +25742,7 @@ with pkgs;
 
   x265 = callPackage ../development/libraries/x265 { };
 
-  xandikos = callPackage ../servers/xandikos { };
+  xandikos = python3.pkgs.callPackage ../servers/xandikos { };
```
Comment (wegank) [inline pkgs/top-level/all-packages.nix:25745]: `callPackage` should be kept for [RFC 140](https://github.com/NixOS/rfcs/blob/master/rfcs/0140-simple-package-paths.md).

**Pull request merged**